docs/explanation/nova-data-migration.rst

```diff
@@ -0,0 +1,85 @@
+==========================================
+Data Migration on nova-cloud-controller
+==========================================
+
+This document explains the details of the database migration for nova-cloud-controller.
+
+nova-cloud-controller Database Migration Details
+------------------------------------------------
+
+When users upgrade the Juju application nova-cloud-controller, the commands **nova-manage db sync** and **nova-manage db online_data_migration** will be executed.
+
+* **nova-manage db sync** - Upgrades the main database schema to the most recent version.
+* **nova-manage db online_data_migration** - Performs data migration to update all live data.
+
+db sync
+~~~~~~~
+
+Before the Yoga version, nova-cloud-controller used SQLAlchemy to handle database migrations. Starting from Yoga, it switched to Alembic, a migrations tool for SQLAlchemy.
+
+Below is a table showing the migration versions from Ussuri to Caracal.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Version
+     - Migration Version
+   * - Ussuri
+     - 407
+   * - Victoria
+     - 412
+   * - Wallaby
+     - 417
+   * - Xena
+     - 422
+   * - Yoga
+     - 422
+       - (move to Alembic)
+       - 8f2f1571d55b_initial_version
+       - 16f1fbcab42b_resolve_shadow_table_diffs
```

Comment:
These should have code formatting, both to keep spellcheck happy, and to be clear they are preformatted identifiers.

```suggestion
       - ``8f2f1571d55b_initial_version``
       - ``16f1fbcab42b_resolve_shadow_table_diffs``
```